sale_order_line_copy/models/sale_order_line.py

```diff
@@ -0,0 +1,15 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, _
+from odoo.exceptions import UserError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def copy_sol(self):
+        for line in self:
+            if line.order_id.state in["cancel", "done"]:
```

Comment:
Missing space (code decoration required by others like OCA):
```suggestion
            if line.order_id.state in ["cancel", "done"]:
```